src/kagglehub/http_resolver.py

```diff
@@ -210,46 +210,44 @@ def is_supported(self, *_, **__) -> bool:  # noqa: ANN002, ANN003
     def __call__(self, h: NotebookHandle, path: Optional[str] = None, *, force_download: Optional[bool] = False) -> str:
         api_client = KaggleApiV1Client()
 
-        cached_response = load_from_cache(h, path)
-        if cached_response and not force_download:
-            return cached_response  # Already cached
-        elif cached_response and force_download:
+        if not h.is_versioned():
+            h.version = _get_current_version(api_client, h)
+
+        nb_path = load_from_cache(h, path)
+        if nb_path and not force_download:
+            return nb_path  # Already cached
+        elif nb_path and force_download:
             delete_from_cache(h, path)
 
-        download_url_root = f"kernels/output/download/{h.owner}/{h.notebook}"
-        output_root = Path(get_cached_path(h, path))
-
-        # List the files and decide how to download them:
-        # - <= 25 files: Download files in parallel
-        # > 25 files: Download the archive and uncompress
-        (files, has_more) = self._list_files(api_client, h) if not path else ([path], False)
-        if has_more:
-            # TODO(b/379761520): add support for .tar.gz archived downloads
-            logger.warning(
-                f"Too many files in {h} (capped at {MAX_NUM_FILES_DIRECT_DOWNLOAD}). "
-                "Unable to download notebook output."
-            )
-            return ""
-
-        # Download files individually in parallel
-        def _inner_download_file(filepath: str) -> None:
-            download_url_path = f"{download_url_root}/{filepath}"
-            full_output_filepath = output_root / filepath
-
-            os.makedirs(os.path.dirname(full_output_filepath), exist_ok=True)
-            api_client.download_file(download_url_path, str(full_output_filepath), h)
-
-        thread_map(
-            _inner_download_file,
-            files,
-            desc=f"Downloading {len(files)} files",
-            max_workers=8,  # Never use more than 8 threads in parallel to download files.
-        )
+        url_path = _build_notebook_download_url_path(h)
+        if h.is_versioned():
+            url_path += f"?version_number={h.version}"
```

Comment:
nit: My earlier comment still stands though it's totally fine to punt on that here.  But with that mindset IMO we should follow the existing patterns like for datasets, specifically we know that `h.is_versioned()` is true here since you filled it above, and we should just push the `h.version` logic into `_build_notebook_download_url_path` itself similar to how other datasources are doing it in this file.